src/github.com/getlantern/detour/detour.go

```diff
@@ -369,60 +365,15 @@ var nonIdempotentMethods = [][]byte{
 	[]byte("PATCH "),
 }
 
-// ref section 9.1.2 of https://www.ietf.org/rfc/rfc2616.txt.
-// checks against non-idemponent methods actually,
-// as we consider the https handshake phase to be idemponent.
-func (dc *Conn) isIdempotentRequest() bool {
-	dc.muLocalBuffer.Lock()
-	defer dc.muLocalBuffer.Unlock()
-	b := dc.localBuffer.Bytes()
+// Ref section 9.1.2 of https://www.ietf.org/rfc/rfc2616.txt.
+// We consider the https handshake phase to be idemponent.
+func isNonIdempotentHTTPRequest(b []byte) bool {
```

Comment:
I am not sure if we can just assume this non-idempotence, as far as I can tell idempotence depends on the application. According to this function GET, DELETE and PUT are considered equally idempotent, but that may not be the case.

For instance, some applications may do things like:

GET /posts/delete?id=123

Which is a GET and by this check would be considered idempotent, nothing prevents a developer for doing that. Is there any way we can stay in the safe side? It seems like this could cause side effects that may be difficult to track back to Lantern.